modules/library/api/src/main/java/org/geotools/styling/StyleFactory.java

```diff
@@ -266,11 +266,14 @@ public RasterSymbolizer createRasterSymbolizer(
 
     public ContrastEnhancement createContrastEnhancement(Expression gammaValue);
 
+    public SelectedChannelType createSelectedChannelType(
+            Expression name, ContrastEnhancement enhancement);
+
     public SelectedChannelType createSelectedChannelType(
             String name, ContrastEnhancement enhancement);
 
     /** @deprecated Use {@link #createSelectedChannelType(String, ContrastEnhancement)} */
-    public SelectedChannelType createSelectedChannelType(String name, Expression gammaValue);
+    public SelectedChannelType createSelectedChannelType(Expression name, Expression gammaValue);
```

Comment:
There should also be a variant createSelectedChannelType(String, Expression), for backwards compatibility, but I don't see it?